ibis/pandas/execution.py

```diff
@@ -461,7 +458,22 @@ def execute_aggregation_dataframe(op, data, scope=None, **kwargs):
         for metric in op.metrics
     ]
 
-    return pd.concat(pieces, axis=1).reset_index()
+    result = pd.concat(pieces, axis=1)
+
+    if op.having:
+        # .having(...) is only accessible on groupby, so this should never
+        # raise
+        assert op.by, (
+            'Filtering out aggregation values is not allowed without at '
+            'least one grouping key'
+        )
+
+        # TODO(phillipc): Don't recompute identical subexpressions
```

Comment:
isn't this (not the nicest syntax in pandas though)

``.groupby(...).filter(....).groupby(..).agg(..)``
?